File: src/main/java/com/andrew121410/mc/ccminecraftbot/packets/handle/pos/OnServerEntityRotationPacket.java
package com.andrew121410.mc.ccminecraftbot.packets.handle.pos;

import com.andrew121410.mc.ccminecraftbot.CCBotMinecraft;
import com.andrew121410.mc.ccminecraftbot.packets.PacketHandler;
import com.andrew121410.mc.ccminecraftbot.player.CCPlayer;
import com.github.steveice10.mc.protocol.packet.ingame.server.entity.ServerEntityRotationPacket;

public class OnServerEntityRotationPacket extends PacketHandler<ServerEntityRotationPacket> {

    @Override
    public void handle(ServerEntityRotationPacket packet, CCBotMinecraft ccBotMinecraft) {
        CCPlayer ccPlayer = ccBotMinecraft.getPlayer();
        if (packet.getEntityId() == ccPlayer.getEntityId()) {
            ccPlayer.getCurrentLocation().add(0, 0, 0, packet.getYaw(), packet.getPitch());
        }
    }
}

File: src/main/kotlin/com/andrew121410/mc/ccminecraftbot/packets/handle/position/OnServerEntityRotationPacket.kt
package com.andrew121410.mc.ccminecraftbot.packets.handle.position

import com.andrew121410.mc.ccminecraftbot.CCBotMinecraft
import com.andrew121410.mc.ccminecraftbot.packets.PacketHandler
import com.github.steveice10.mc.protocol.packet.ingame.server.entity.ServerEntityRotationPacket

class OnServerEntityRotationPacket : PacketHandler<ServerEntityRotationPacket>() {
    override fun handle(packet: ServerEntityRotationPacket, ccBotMinecraft: CCBotMinecraft) {
        val ccPlayer = ccBotMinecraft.player
        if (packet.entityId == ccPlayer.entityId) {
            ccPlayer.currentLocation!!.add(0.0, 0.0, 0.0, packet.yaw, packet.pitch)
        }
    }
}